File: web/buildwise-web/src/Features/MaterialRequests/services/materialRequestService.js
const API_BASE_URL = 'http://localhost:5078/api';

export const materialRequestService = {
  async getRequests(status = '', projectId = '') {
    let url = `${API_BASE_URL}/materialrequests`;
    const params = new URLSearchParams();
    if (status) params.append('status', status);
    if (projectId) params.append('projectId', projectId);
    if (params.toString()) url += `?${params.toString()}`;

    const res = await fetch(url);
    if (!res.ok) throw new Error('Failed to fetch material requests');
    return await res.json();
  },

  async getRequestById(id) {
    const res = await fetch(`${API_BASE_URL}/materialrequests/${id}`);
    if (!res.ok) throw new Error(`Failed to fetch material request #${id}`);
    return await res.json();
  },

  async createRequest(payload) {
    const res = await fetch(`${API_BASE_URL}/materialrequests`, {
      method: 'POST',
      headers: { 'Content-Type': 'application/json' },
      body: JSON.stringify(payload)
    });
    if (!res.ok) {
      const err = await res.text();
      throw new Error(err || 'Failed to create material request');
    }
    return await res.json();
  },

  async submitRequest(id) {
    const res = await fetch(`${API_BASE_URL}/materialrequests/${id}/submit`, {
      method: 'POST'
    });
    if (!res.ok) throw new Error('Failed to submit material request');
    return await res.json();
  },

  async approveRequest(id, payload) {
    const res = await fetch(`${API_BASE_URL}/materialrequests/${id}/approve`, {
      method: 'POST',
      headers: { 'Content-Type': 'application/json' },
      body: JSON.stringify(payload)
    });
    if (!res.ok) throw new Error('Failed to record approval decision');
    return await res.json();
  },

  async runPlanningAgent(id) {
    const res = await fetch(`${API_BASE_URL}/materialrequests/${id}/plan`, {
      method: 'POST'
    });
    if (!res.ok) throw new Error('Failed to execute Procurement Planning Agent AI');
    return await res.json();
  }
};

File: mobile/buildwise_mobile/lib/features/material_requests/services/material_request_service.dart
import 'dart:convert';
import 'package:http/http.dart' as http;

class MaterialRequestService {
  static const String baseUrl = 'http://localhost:5078/api/materialrequests';

  Future<List<dynamic>> getRequests() async {
    final response = await http.get(Uri.parse(baseUrl));
    if (response.statusCode == 200) {
      return json.decode(response.body);
    } else {
      throw Exception('Failed to load material requests');
    }
  }

  Future<void> createRequest(Map<String, dynamic> data) async {
    final response = await http.post(
      Uri.parse(baseUrl),
      headers: {'Content-Type': 'application/json'},
      body: json.encode(data),
    );
    if (response.statusCode != 201 && response.statusCode != 200) {
      throw Exception('Failed to create material request');
    }
  }
}
